includes/download.clip.inc

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * @file
+ * Defines menu callback for downloading a clip of the given image.
+ */
+
+/**
+ * Menu callback downloads the given clip.
+ *
+ * @param AbstractObject $object
+ *   An AbstractObject representing an object within Fedora.
+ */
+function islandora_openseadragon_download_clip(AbstractObject $object) {
+  if (isset($_GET['clip'])) {
+    module_load_include('inc', 'islandora_openseadragon', 'includes/utilities');
+    $clip_parts = islandora_openseadragon_construct_clip_url($_GET['clip'], TRUE);
+    if ($clip_parts) {
+      $filename = $object->label;
+      header("Content-Disposition: attachment; filename=\"{$filename}.jpg\"");
+      header("Content-type: image/jpeg");
+      header("Content-Transfer-Encoding: binary");
+      $ch = curl_init();
+      curl_setopt($ch, CURLOPT_BINARYTRANSFER, 1);
+      curl_setopt($ch, CURLOPT_HEADER, 0);
+      curl_setopt($ch, CURLOPT_URL, $clip_parts['image_url']);
+      curl_exec($ch);
+      curl_close($ch);
+    }
+    else {
+      drupal_access_denied();
+      watchdog('islandora_openseadragon', 'Invalid parameters specified for downloading of a clip for @pid. Parameters attempted: @params.', array(
+        '@pid' => $object->id,
+        '@params' => $_GET['clip'],
+      ));
+    }
+  }
+  exit();
```

Comment:
I think this is the only use of `exit()` in the module, and this code block was just a move rather than new functionality / changes. That being said using `drupal_exit()` seems like a safe enough change, unlikely to introduce problems.